fs/rails-base-graphql-api
# Password recovery emails

### Summary

Allows user to request password recovery.
Partially resolves #18 

### Test plan

List of steps to manually test introduced functionality:

* Open [GraphiQL Interface](http://localhost:3000/graphiql)
* Make request using schema:
```graphql
mutation RecoverPassword($email: String!) {
  requestPasswordRecovery(email: $email) {
    message
    detail
  }
}
```
* In development your browser should open recovery letter
* In production you should receive this email

-------------

GraphiQL query example
![Query ](https://user-images.githubusercontent.com/57901245/83756001-101ee280-a677-11ea-8f26-4d56afbe5751.png)

Letter example
![image](https://user-images.githubusercontent.com/57901245/83756037-1ca33b00-a677-11ea-9e1b-c1ee9304d670.png)

-------------

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Provide with env variables `MAILER_SENDER_ADDRESS`, `SENDGRID_USERNAME` and `SENDGRID_PASSWORD`.
Run `bin/rails db:migrate`.
